src/main/java/inf112/skeleton/app/Game.java

```diff
@@ -54,29 +56,55 @@ public void pause() {
     public void resume() {
     }
 
-    /*
-     * @keyUp makes it feel laggy, we're using @keyDown for immediate response.
-     * This won't be a problem because it's not triggered on pressed keys.
-     */
     @Override
     public boolean keyDown(int key) {
-        int x = 0, y = 0;
+        Vector2 deltaPos = new Vector2(0,0);
         switch (key){
-            case Input.Keys.RIGHT: x++; break;
-            case Input.Keys.LEFT: x--; break;
-            case Input.Keys.UP: y++; break;
-            case Input.Keys.DOWN: y--; break;
+            case Input.Keys.RIGHT: deltaPos.x++; break;
+            case Input.Keys.LEFT: deltaPos.x--; break;
+            case Input.Keys.UP: deltaPos.y++; break;
+            case Input.Keys.DOWN: deltaPos.y--; break;
+
+            // movement via cards
+            case Input.Keys.W: checkCardKind(new Card(CardKind.FORWARD, 2, 0)); break;
+            case Input.Keys.S: checkCardKind(new Card(CardKind.REVERSE, 2, 0)); break;
+            case Input.Keys.D: checkCardKind(new Card(CardKind.TURNRIGHT, 1, 0)); break;
+            case Input.Keys.A: checkCardKind(new Card(CardKind.TURNLEFT, 1, 0)); break;
+            case Input.Keys.F: checkCardKind(new Card(CardKind.FLIP, 2, 0)); break;
+
         }
-        movePlayer(x, y);
+        movePlayer(deltaPos);
         return true;
     }
 
-    /*
-     * I'm not checking if the player goes outside of the board
-     * because it's a bad idea to write code before we know it's needed.
+    /**
+     * checks the action of given card
+     * @param card card to check
      */
-    private void movePlayer(int x, int y){
-        playerPos.add((float)x, (float)y);
+    private void checkCardKind(Card card) {
```

Comment:
"check..." høres ut som om det bare sjekker noe isteden for å utføre noe, så jeg ville ha skrevet "execute", etc.
"...CardKind" blir også litt rart siden den tar et Card som input, så jeg ville ha kalt metoden for "executeCard".